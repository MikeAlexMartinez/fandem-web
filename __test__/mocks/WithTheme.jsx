import React from 'react';
import { ThemeProvider } from '@material-ui/styles';
import muiTheme from '../../styles/mui-theme';

const WithTheme = ({ children }) => (
  <ThemeProvider theme={muiTheme}>
    {children}
  </ThemeProvider>
);

export default WithTheme;
