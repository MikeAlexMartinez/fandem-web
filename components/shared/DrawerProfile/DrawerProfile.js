import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import CurrentUser from "../CurrentUser";

const DrawerProfile = ({ classes }) => (
  <CurrentUser>
    {({ data: { currentUser }, loading, error }) => {
      console.log(currentUser);
      return (
        <div className={classNames(classes.userProfile)}>
          <div>User Profile</div>
        </div>
      );
    }}
  </CurrentUser>
);

DrawerProfile.propTypes = {
  classes: PropTypes.object
};

export default DrawerProfile;
