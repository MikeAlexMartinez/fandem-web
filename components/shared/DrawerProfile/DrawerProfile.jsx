import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Link from 'next/link';
import { Typography, CircularProgress, Fab } from '@material-ui/core';
import { Edit } from '@material-ui/icons';

import CurrentUser from '../CurrentUser';
import ErrorMessage from '../ErrorMessage/ErrorMessage';

const DrawerProfile = ({ classes }) => (
  <CurrentUser>
    {({ data: { currentUser }, loading, error }) => {
      if (error) {
        return <ErrorMessage message="Error retrieving user information" />;
      }

      if (loading) {
        return (
          <div
            className={classNames(
              classes.spinnerContainer,
              'flex row jc-center ai-center',
            )}
          >
            <CircularProgress size={80} />
          </div>
        );
      }

      const profilePicture = currentUser
        && currentUser.profilePicture
        && currentUser.profilePicture.length === 1
        && currentUser.profilePicture[0].photo;
      return (
        <div
          className={classNames(
            classes.userProfile,
            'flex column jc-start ai-stretch',
          )}
        >
          {/* picture, edit button & notifs */}
          <div className={classNames('flex row jc-sb ai-start')}>
            <div className={classNames(classes.imgContainer)}>
              {!profilePicture.image && (
                <img
                  alt="profile placeholder"
                  src="/static/images/male-profile-image.png"
                  className={classNames(classes.photo)}
                />
              )}
              {profilePicture.image && (
                <img
                  alt="profile"
                  src={profilePicture.image}
                  className={classNames(classes.photo)}
                />
              )}
            </div>
            {/* Actions */}
            <div
              className={classNames(
                classes.actions,
                'flex column jc-start ai-end',
              )}
            >
              <Fab>
                <Link href="/editprofile">
                  <Edit />
                </Link>
              </Fab>
            </div>
          </div>
          {/* greeting */}
          <div
            className={classNames(
              classes.greeting,
              'flex row jc-center ai-center',
            )}
          >
            <Typography
              variant="subtitle2"
              className={classNames(classes.greetingText)}
            >
              Hello
              {' '}
              {currentUser.name}
            </Typography>
          </div>
        </div>
      );
    }}
  </CurrentUser>
);

DrawerProfile.propTypes = {
  classes: PropTypes.object,
};

export default DrawerProfile;
