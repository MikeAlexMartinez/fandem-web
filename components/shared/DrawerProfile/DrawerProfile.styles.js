export default function(theme) {
  return {
    userProfile: {
      height: "250px",
      backgroundColor: theme.palette.secondary.light
    }
  };
}
