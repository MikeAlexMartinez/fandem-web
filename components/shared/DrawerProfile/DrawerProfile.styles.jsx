export default function styles(theme) {
  return {
    spinnerContainer: {
      height: '250px',
      padding: '50px',
    },
    userProfile: {
      height: '250px',
      backgroundColor: theme.palette.secondary.light,
    },
    imgContainer: {
      overflow: 'hidden',
      height: '180px',
      width: '180px',
      backgroundColor: 'white',
      borderRadius: '0 50% 50% 0',
    },
    photo: {
      height: '180px',
      width: '100%',
      objectFit: 'cover',
    },
    greeting: {
      height: '70px',
    },
    greetingText: {
      fontSize: '1.5rem',
      color: theme.palette.grey[50],
    },
    actions: {
      padding: '1rem',
    },
  };
}
