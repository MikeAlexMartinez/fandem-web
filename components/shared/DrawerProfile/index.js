import { withStyles } from "@material-ui/core";

import DrawerProfile from "./DrawerProfile";
import styles from "./DrawerProfile.styles";

export default withStyles(styles)(DrawerProfile);
