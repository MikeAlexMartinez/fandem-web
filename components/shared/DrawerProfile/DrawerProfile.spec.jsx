import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import WithTheme from '../../../__test__/mocks/WithTheme';

import DrawerProfile from './DrawerProfile';

describe('<DrawerProfile />', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <WithTheme>
          <DrawerProfile />
        </WithTheme>
      </MockedProvider>,
    );

    const drawerProfile = wrapper.find('DrawerProfile');

    expect(toJSON(drawerProfile)).toMatchSnapshot();
  });
});
