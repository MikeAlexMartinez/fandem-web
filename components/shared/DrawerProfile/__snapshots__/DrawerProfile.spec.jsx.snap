// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerProfile /> renders and matches snapshot 1`] = `
<DrawerProfile
  classes={
    Object {
      "actions": "DrawerProfile-actions-7",
      "greeting": "DrawerProfile-greeting-5",
      "greetingText": "DrawerProfile-greetingText-6",
      "imgContainer": "DrawerProfile-imgContainer-3",
      "photo": "DrawerProfile-photo-4",
      "spinnerContainer": "DrawerProfile-spinnerContainer-1",
      "userProfile": "DrawerProfile-userProfile-2",
    }
  }
>
  <CurrentUser>
    <Query
      query={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "CURRENT_USER_QUERY",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "currentUser",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "email",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "name",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "countryCode",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "phoneNumber",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "favoriteTeam",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": Object {
                                  "kind": "Name",
                                  "value": "label",
                                },
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": Object {
                                  "kind": "Name",
                                  "value": "value",
                                },
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "shortName",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "country",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": Object {
                                  "kind": "Name",
                                  "value": "label",
                                },
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": Object {
                                  "kind": "Name",
                                  "value": "value",
                                },
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                        Object {
                          "directives": Array [],
                          "kind": "FragmentSpread",
                          "name": Object {
                            "kind": "Name",
                            "value": "ProfilePicture",
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "displayName",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "isPrivate",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "emailValidated",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "subscriptions",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "userRoles",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "followers",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "followers",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "user",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "displayName",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "isPrivate",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "directives": Array [],
                                            "kind": "FragmentSpread",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "ProfilePicture",
                                            },
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "influencers",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "influencers",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "user",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "displayName",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "isPrivate",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "directives": Array [],
                                            "kind": "FragmentSpread",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "ProfilePicture",
                                            },
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
            Object {
              "directives": Array [],
              "kind": "FragmentDefinition",
              "name": Object {
                "kind": "Name",
                "value": "ProfilePicture",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": Object {
                      "kind": "Name",
                      "value": "profilePicture",
                    },
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "where",
                        },
                        "value": Object {
                          "fields": Array [
                            Object {
                              "kind": "ObjectField",
                              "name": Object {
                                "kind": "Name",
                                "value": "isProfile",
                              },
                              "value": Object {
                                "kind": "BooleanValue",
                                "value": true,
                              },
                            },
                          ],
                          "kind": "ObjectValue",
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "photos",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "photo",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "image",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "typeCondition": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "User",
                },
              },
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 949,
            "start": 0,
          },
        }
      }
    >
      <div
        className="DrawerProfile-spinnerContainer-1 flex row jc-center ai-center"
      >
        <WithStyles(ForwardRef(CircularProgress))
          size={80}
        >
          <ForwardRef(CircularProgress)
            classes={
              Object {
                "circle": "MuiCircularProgress-circle",
                "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                "circleStatic": "MuiCircularProgress-circleStatic",
                "colorPrimary": "MuiCircularProgress-colorPrimary",
                "colorSecondary": "MuiCircularProgress-colorSecondary",
                "indeterminate": "MuiCircularProgress-indeterminate",
                "root": "MuiCircularProgress-root",
                "static": "MuiCircularProgress-static",
                "svg": "MuiCircularProgress-svg",
              }
            }
            size={80}
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 80,
                  "width": 80,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </ForwardRef(CircularProgress)>
        </WithStyles(ForwardRef(CircularProgress))>
      </div>
    </Query>
  </CurrentUser>
</DrawerProfile>
`;
